containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,279 @@
+                try {
+                    Thread.sleep(2000);
+                } catch (InterruptedException e1) {
+                    e1.printStackTrace();
+                }
+            }
+        } while (!connected);
+        log.debug("Connection to Cassandra establised");
+
+        // ensure that the openshift_metrics keyspace exists
+        ensureKeyspace("openshift_metrics");
+
+        // if we already have a mappingTable stored in OpenShift, then use it.
+        // Otherwise we need to generate the overwrite table.
+        if (hasMappingTable()) {
+            log.debug("Mapping table already exists in Cassandra. Using this table");
+            this.overrides = getMappingTable();
+        } else {
+            log.debug("Mapping table does not exist in Cassandra. Creating Mapping");
+            // get the current list of tenants in Hawkular Metrics
+            Set<String> tenants = getTenants();
+
+            // map the current tenants to the current projects
+            Map<String, String> namespaceMapping = getNamespaceMapping(tenants);
+            log.debug("Namespace mapping: " + namespaceMapping);
+
+            this.overrides = namespaceMapping;
+
+            // write this mapping table back to Cassandra. We don't want to ever recalculate this table
+            log.debug("Writing the mapping table to Cassandra");
+            writeMappingTable(this.overrides);
+            log.debug("Mapping table stored in Cassandra");
+        }
+    }
+
+    private Session createSession() {
+        Cluster.Builder clusterBuilder = new Cluster.Builder();
+
+        String nodes = System.getProperty("hawkular.metrics.cassandra.nodes", "hawkular-cassandra");
+        Arrays.stream(nodes.split(",")).forEach(clusterBuilder::addContactPoint);
+
+        if (System.getProperty("hawkular.metrics.cassandra.use-ssl") != null && !System.getProperty("hawkular.metrics.cassandra.use-ssl").equals("false")) {
+            SSLOptions sslOptions = null;
+            try {
+                String[] defaultCipherSuites = {"TLS_RSA_WITH_AES_128_CBC_SHA", "TLS_RSA_WITH_AES_256_CBC_SHA"};
+                sslOptions = JdkSSLOptions.builder().withSSLContext(SSLContext.getDefault())
+                        .withCipherSuites(defaultCipherSuites).build();
+                clusterBuilder.withSSL(sslOptions);
+            } catch (NoSuchAlgorithmException e) {
+                throw new RuntimeException("SSL support is required but is not available in the JVM.", e);
+            }
+        }
+
+        Cluster cluster = clusterBuilder.build();
+        cluster.init();
+
+        Session session = cluster.connect();
+
+        return session;
+    }
+
+    /**
+     * Creates a keyspace if it does not already exist
+     * @param keyspace The name of the keyspace
+     */
+    private void ensureKeyspace(String keyspace) {
+        ResultSet resultset = session.execute("SELECT * FROM system_schema.keyspaces WHERE keyspace_name = '" + keyspace + "';");
```

Comment:
You can simply this method. You do not need to query system_schema.keyspaces. Use the ```IF NOT EXISTS``` clause, e.g.,

```
CREATE KEYSPACE IF NOT EXISTS foo WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}
```